store/sqlstore/draft_store.go

```diff
@@ -0,0 +1,244 @@
+	return []interface{}{
+		draft.CreateAt,
+		draft.UpdateAt,
+		draft.DeleteAt,
+		draft.Message,
+		draft.RootId,
+		draft.ChannelId,
+		draft.UserId,
+		model.ArrayToJSON(draft.FileIds),
+		model.StringInterfaceToJSON(draft.Props),
+	}
+}
+
+func newSqlDraftStore(sqlStore *SqlStore, metrics einterfaces.MetricsInterface) store.DraftStore {
+	return &SqlDraftStore{
+		SqlStore:           sqlStore,
+		metrics:            metrics,
+		maxDraftSizeCached: model.PostMessageMaxRunesV1,
+	}
+}
+
+func (s *SqlDraftStore) Get(userId, channelId, rootId string) (*model.Draft, error) {
+	query := s.getQueryBuilder().
+		Select("*").
+		From("Drafts").
+		Where(sq.Eq{
+			"UserId":    userId,
+			"ChannelId": channelId,
+			"RootId":    rootId,
+		})
+
+	dt := model.Draft{}
+	err := s.GetReplicaX().GetBuilder(&dt, query)
+
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return nil, store.NewErrNotFound("Draft", channelId)
+		}
+		return nil, errors.Wrapf(err, "failed to find draft with channelid = %s", channelId)
+	}
+
+	return &dt, nil
+}
+
+func (s *SqlDraftStore) Save(draft *model.Draft) (*model.Draft, error) {
+	draft.PreSave()
+	maxDraftSize := s.GetMaxDraftSize()
+	if err := draft.IsValid(maxDraftSize); err != nil {
+		return nil, err
+	}
+
+	builder := s.getQueryBuilder().Insert("Drafts").Columns(draftSliceColumns()...).Values(draftToSlice(draft)...)
+	query, args, err := builder.ToSql()
+
+	if err != nil {
+		return nil, errors.Wrap(err, "save_draft_tosql")
+	}
+
+	if _, err = s.GetMasterX().Exec(query, args...); err != nil {
+		return nil, errors.Wrap(err, "failed to save Draft")
+	}
+
+	return draft, nil
+}
+
+func (s *SqlDraftStore) Update(draft *model.Draft) (*model.Draft, error) {
+	draft.PreUpdate()
+
+	maxDraftSize := s.GetMaxDraftSize()
+	if err := draft.IsValid(maxDraftSize); err != nil {
+		return nil, err
+	}
+
+	query := s.getQueryBuilder().
+		Update("Drafts").
+		Set("CreateAt", draft.CreateAt).
+		Set("UpdateAt", draft.UpdateAt).
+		Set("DeleteAt", draft.DeleteAt).
+		Set("UserId", draft.UserId).
+		Set("ChannelId", draft.ChannelId).
+		Set("RootId", draft.RootId).
+		Set("Message", draft.Message).
+		Set("Props", draft.Props).
+		Set("FileIds", draft.FileIds).
+		Where(sq.Eq{
+			"UserId":    draft.UserId,
+			"ChannelId": draft.ChannelId,
+			"RootId":    draft.RootId,
+		})
+
+	sql, args, err := query.ToSql()
+
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to convert to sql")
+	}
+
+	if _, err = s.GetMasterX().Exec(sql, args...); err != nil {
+		return nil, errors.Wrapf(err, "failed to update Draft with channelid=%s", draft.ChannelId)
+	}
+
+	return draft, nil
+}
+
+func (s *SqlDraftStore) GetDraftsForUser(userID, teamID string) ([]*model.Draft, error) {
+	var drafts []*model.Draft
+
+	query := s.getQueryBuilder().
+		Select("Drafts.*").
+		From("Drafts").
+		InnerJoin("ChannelMembers ON ChannelMembers.ChannelId = Drafts.ChannelId").
+		Where(sq.And{
+			sq.NotEq{"Drafts.Message": ""},
```

Comment:
Is it a valid case that the `Message` can be empty in DB?

Reply:
It is a valid case, as there could be drafts with just file attachments, but good catch as this was blocking those drafts from being fetched. Removing this line :+1: